### PR TITLE
#108 arrays support

### PR DESCRIPTION
**Pull request by serebrserg**

### CODE REVIEW

File: src/test/java/ru/yandex/clickhouse/integration/RowBinaryStreamTest.java [hunk 1]
```diff
@@ -150,6 +206,77 @@ public void writeTo(ClickHouseRowBinaryStream stream) throws IOException {
         Assert.assertEquals(rs.getDouble("float32"), 123.456);
         Assert.assertEquals(rs.getDouble("float64"), 42.21);
 
+        final Date[] dateArray = (Date[]) rs.getArray("dateArray").getArray();
+        Assert.assertEquals(dateArray.length, dates1.length);
+        for (int i = 0; i < dateArray.length; i++) {
+            // expected is Date at start of the day in server timezone
+            DateTime dt = new DateTime(dates1[i].getTime())
```
Comment (serebrserg) [inline src/test/java/ru/yandex/clickhouse/integration/RowBinaryStreamTest.java]: @genuss , Regarding the tests failure I mentioned in https://github.com/yandex/clickhouse-jdbc/pull/108,  the code actually worked correctly, the problem I found was in this test.
Expected Date value from Date field must be at start of the day in server timezone. Please check if I've done it correctly.
Also I've reproduced this at CI by setting timezone to ClickhHouse server different from UTC (that was enough) and  from client application(tests).

Comment (genuss) [inline src/test/java/ru/yandex/clickhouse/integration/RowBinaryStreamTest.java]: Your suggestion looks right. Wouldn't it be neccessarry to check wih clickhouse-server developers?

Comment (genuss) [inline src/test/java/ru/yandex/clickhouse/integration/RowBinaryStreamTest.java]: Any news here? @serebrserg

Comment (serebrserg) [inline src/test/java/ru/yandex/clickhouse/integration/RowBinaryStreamTest.java]: Yes, in fact having discussed this with clickhouse developers and checked the documentation and implementations of other drivers, I consider that the current behavior with dates that I described in my previous comment incorrect.
I have a PR to fix that https://github.com/yandex/clickhouse-jdbc/pull/120
You can make a quick review of it if you'd like. I plan to merge it in a couple of days.
Then the tests in this PR need to be fixed.

Comment (genuss) [inline src/test/java/ru/yandex/clickhouse/integration/RowBinaryStreamTest.java]: Your solution with ``use_server_time_zone_for_dates`` looks the most flexible. Looking forward to have it it master!

**Pull request merged**